lib/client/convertIssueDataToDetailedViewGroup.ts

```diff
@@ -0,0 +1,88 @@
+import { State } from '@hookstate/core';
+import { group } from 'd3';
+import { groupBy, reverse, sortBy, uniqBy } from 'lodash';
+
+import { ViewMode } from '../enums';
+import {getLinkForRoadmapChild} from './linkUtils';
+import { DetailedViewGroup, IssueData } from '../types';
+
+function flattenIssueData(issueData: IssueData, isChildIssue = false): IssueData[] {
+  const parentArray: IssueData[] = []
+  if (isChildIssue) {
+    parentArray.push(issueData)
+  }
+  const childrenArray = issueData.children.flatMap((child) => flattenIssueData(child, true))
+  return uniqBy(parentArray.concat(childrenArray), 'html_url')
+}
+
+export function convertIssueDataStateToDetailedViewGroupOld(issueDataState: State<IssueData>, viewMode: ViewMode): DetailedViewGroup[] {
+  const newIssueData = issueDataState.children.value.map((v) => ({
+    ...v,
+    group: v.parent?.title ?? '',
+    children: v.children.map((x) => ({ ...x, group: x.parent?.title ?? '' })),
+  }));
+
+  const getGroupedIssues = (issueData: IssueData[]): DetailedViewGroup[] => Array.from(
+    group(issueData, (d) => d.group),
+    ([key, value]) => ({
+      groupName: key,
+      items: value,
+      url: getLinkForRoadmapChild(newIssueData.find((i) => i.title === key)),
+    }),
+  );
+
+  const issueDataLevelOneIfNoChildren: IssueData[] = newIssueData.map((v) => ({ ...v, children: [v], group: v.title }));
+
+  const issueDataLevelOne: IssueData[] = newIssueData.map((v) => v.children.flat()).flat();
+  const issueDataLevelOneGrouped: DetailedViewGroup[] = getGroupedIssues(issueDataLevelOne);
+  const issueDataLevelOneIfNoChildrenGrouped: DetailedViewGroup[] = getGroupedIssues(issueDataLevelOneIfNoChildren);
+
+  let issuesGrouped: DetailedViewGroup[];
+  if (viewMode === ViewMode.Detail) {
+    issuesGrouped =
+      (!!issueDataLevelOneGrouped && issueDataLevelOneGrouped.length > 0 && issueDataLevelOneGrouped) ||
```

Comment:
no idea. I just copied over the original code that was in the roadmap component that Alex wrote a while ago because the ones I attempted to replace with weren't parsing things exactly how we needed. This is why I added the fixtures, so we can test against real roadmaps and ensure we're getting the exact results we want.

Reply:
fixed it in updates